internal/otelutil/otel_tracing.go

```diff
@@ -97,7 +97,8 @@ func SetupTracing(cfg config.OTelTracingConfig, resourceLibrary string, resource
 			otelconfig.WithTracesEnabled(true),
 			otelconfig.WithSampler(samplers.TraceIDRatioBased(sampleRatio)),
 			otelconfig.WithHeaders(map[string]string{
-				"x-honeycomb-team": cfg.APIKey,
+				"x-honeycomb-team":    cfg.APIKey,
+				"x-honeycomb-dataset": cfg.Dataset,
```

Comment:
We should probably check the API key length and only set this header when it's a classic key. We did this in our distros - https://github.com/honeycombio/honeycomb-opentelemetry-go/blob/f93016685160d16fd11b0831db80396038fb1b1c/honeycomb.go#L215